src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -0,0 +1,107 @@
+        public string FallbackHref { get; set; }
+
+        /// <summary>
+        /// The class name defined in the stylesheet to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestClassAttributeName)]
+        public string FallbackTestClass { get; set; }
+
+        /// <summary>
+        /// The CSS property name to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestPropertyAttributeName)]
+        public string FallbackTestProperty { get; set; }
+
+        /// <summary>
+        /// The CSS property value to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestValueAttributeName)]
+        public string FallbackTestValue { get; set; }
+
+        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
+        // TODO: Change this to activate ILogger<LinkTagHelper> directly once https://github.com/aspnet/Hosting/pull/147 is in
+        [Activate]
+        protected internal ILoggerFactory LoggerFactory { get; set; }
+
+        /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            var logger = LoggerFactory.Create<LinkTagHelper>();
+
+            if (!context.ShouldProcess(RequiredAttributes, logger))
+            {
+                // TODO: Should we trace the current view/request path here too?
+                logger.WriteVerbose("Skipping processing for {0} {1}", nameof(LinkTagHelper), context.UniqueId);
+                return;
+            }
+
+            var content = new StringBuilder();
+
+            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+
+            // We've taken over rendering here so prevent the element rendering the outer tag
+            output.TagName = null;
+
+            // Rebuild the <link /> tag that loads the primary stylesheet
+            content.Append("<link ");
+            foreach (var a in output.Attributes)
+            {
+                content.AppendFormat(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", a.Key, a.Value);
+            }
+            content.AppendLine("/>");
+
+            // Build the <meta /> tag that's used to test for the presence of the stylesheet
+            content.AppendLine(string.Format(CultureInfo.InvariantCulture, FallbackTestMetaTemplate, FallbackTestClass));
+
+            // Build the <script /> tag that checks the effective style of <meta /> tag above and renders the extra
+            // <link /> tag to load the fallback stylesheet if the test CSS property value is found to be false,
+            // indicating that the primary stylesheet failed to load.
+            content.Append("<script>");
+            content.AppendFormat(CultureInfo.InvariantCulture,
+                                 JavaScriptHelpers.GetEmbeddedJavaScript(FallbackJavaScriptResourceName, isFormatString: true),
```

Comment:
Do we typically inject readable `JavaScript`? Since it's auto generated I'd think it'd be best to minify it or at least allow an option to minify it.